# EPICON.Mycobiome

The codes were used in the analysis of Fungal community data generated from EPICON project.

EPICON is a DOE funded project aimed at understanding responses of sorghum to drought, from host gene expression to host-associated microbiome.



For detailed descriptions of experimental design, please see associated publications:

            Gao et al Fungal community assembly: Drift, selection, drought-stress, and universal ecological dynamics
      
            Gao et al Strong succession in arbuscular mycorrhizal fungal communities ISMEJ 2019 https://www.nature.com/articles/s41396-018-0264-0 
      
            Xu et al Drought delays development of the sorghum root microbiome and enriches for monoderm bacteria PNAS 2018 https://www.pnas.org/content/115/18/E4284




Briefly, we collected 1026 sample from three treatments, two sorghum culrivars, 17 time points and four compartments.

            Three treatments: control, pre-flowering drought and post-flowering drought
      
            Two sorghum cultivars: the pre-flowering, drought tolerant sorghum cultivar RTx430, and the post-flowering, drought tolerant (or ‘stay green’) cultivar BTx642
      
            Four compartments: Leaf, Root, Rhizosphere, Soil



Fungal community compositon and abundance was recognized by:

            Illumina Miseq sequencing of fungal ITS2 rDNA region using 5.8SFun and ITS4Fun primers of the leaf, root, rhizosphere and soil
      
            Illumina Miseq sequencing of fungal ITS2 region for 26 air samples.
      
            qPCR of fungal 18S rRNA region using FF2 and FR1 primers of leaf and root samples
      
            Fungal transcripts in transcriptomes containing both sorghum plants and their fungi generated from leaf and root samples
      


Fungal community assembly was explored by:

            Multivariate analysis such as PCoA, adonis, betadispersion
      
            Raup-Crick index and Beta Nearest Taxon Index 
      
            Dissimilarity Overlap Curve
      
            etc...



